### PR TITLE
PDOException extends RuntimeException and can use int code errors

### PR DESCRIPTION
**Pull request by VincentLanglet**
Closes https://github.com/vimeo/psalm/issues/7672

### CODE REVIEW

File: src/Psalm/Internal/Analyzer/Statements/Expression/Call/Method/MethodCallReturnTypeFetcher.php [hunk 1]
```diff
@@ -97,16 +98,14 @@ public static function fetch(
 
         if ($premixin_method_id->method_name === 'getcode'
             && $premixin_method_id->fq_class_name !== Exception::class
+            && $premixin_method_id->fq_class_name !== RuntimeException::class
+            && $premixin_method_id->fq_class_name !== PDOException::class
```
Comment (orklah) [inline src/Psalm/Internal/Analyzer/Statements/Expression/Call/Method/MethodCallReturnTypeFetcher.php:102]: Please use "PDOException" as a string. We don't want to introduce a false dependency when it's not needed

Comment (VincentLanglet) [reply src/Psalm/Internal/Analyzer/Statements/Expression/Call/Method/MethodCallReturnTypeFetcher.php:102]: It was already used with `$premixin_method_id->fq_class_name === PDOException::class`, you said it was ok for FQCN: https://github.com/vimeo/psalm/pull/7390#discussion_r784799089

Comment (orklah) [reply src/Psalm/Internal/Analyzer/Statements/Expression/Call/Method/MethodCallReturnTypeFetcher.php:102]: Yeah, but I believe we went back on another PR, let me check that

Comment (orklah) [reply src/Psalm/Internal/Analyzer/Statements/Expression/Call/Method/MethodCallReturnTypeFetcher.php:102]: nvm, this was for constants for allowing to use them without having the extension: https://github.com/vimeo/psalm/pull/7107/files#r765332473

It shouldn't be an issue for ::class, forget I said anything :)

Comment (AndrolGenhald) [reply src/Psalm/Internal/Analyzer/Statements/Expression/Call/Method/MethodCallReturnTypeFetcher.php:102]: It was changed to a string in #7648.

**Comment by orklah** [comment]
@VincentLanglet what's the status of this? It seems to still be in draft on your side

**Comment by VincentLanglet** [comment]
> @VincentLanglet what's the status of this? It seems to still be in draft on your side

The PR is ready and I made the same on the Phpstan. I was just waiting an answer/merge from ondrej on the Phpstan PR before. In order to be sure that both psalm and phpstan will have the same behavior.

**Comment by VincentLanglet** [comment]
Seems like the PR https://github.com/phpstan/phpstan-src/pull/1018 will be accepted on phpstan

**Comment by orklah** [comment]
Thanks!

**Pull request merged**